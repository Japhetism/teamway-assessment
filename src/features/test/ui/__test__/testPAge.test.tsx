import { render, screen, within } from "@testing-library/react";
import { createRenderer } from "react-test-renderer/shallow";
import { RecoilRoot } from "recoil";
import Test from "..";

const renderer = createRenderer();

const defaultComponent = <RecoilRoot><Test /></RecoilRoot>;

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Unit test for test page ", () => {

    it("should render and match the snapshot", () => {
        renderer.render(defaultComponent);
        const renderedOutput = renderer.getRenderOutput();

        expect(renderedOutput).toMatchSnapshot();
    });

    it("should have title as Your result", () => {
        const { getByRole } = render(defaultComponent);
        const titleHeading = getByRole("heading", { name: "Are you an introvert or an extrovert?" });

        expect(titleHeading).toBeVisible();
    });

});
