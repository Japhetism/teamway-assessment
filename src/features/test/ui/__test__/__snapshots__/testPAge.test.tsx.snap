// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for test page  should render and match the snapshot 1`] = `
<RecoilRoot_INTERNAL>
  <Test />
</RecoilRoot_INTERNAL>
`;
